nrf_modem: doc: sockets: add info on CID and CONN_SAVE socket options (#1051)
nrfconnect/sdk-nrfxlib: nrf_modem/doc/sockets.rst

NRF_SO_SEC_DTLS_CID
   Set DTLS Connection ID setting.
   This socket option decides whether the modem will request or accept a DTLS connection ID when performing the server handshake.
   The socket option is supported from modem firmware v1.3.x, where x is greater than or equal to 5, and v2.x.x.
   The following values are accepted:

>    * :c:macro:`NRF_SO_SEC_DTLS_CID_DISABLED` - The connection ID extension is not included in the client hello. Hence, the DTLS connection ID is not used.
>    * :c:macro:`NRF_SO_SEC_DTLS_CID_SUPPORTED` - The connection ID extension with a zero-length CID is included in the client hello. Hence, the modem will accept a DTLS connection ID from the server.
>    * :c:macro:`NRF_SO_SEC_DTLS_CID_ENABLED` - The connection ID extension with a valid CID is included in the client hello. Hence, DTLS connection ID support is requested by the modem.

Reviewer: ```suggestion
   * :c:macro:`NRF_SO_SEC_DTLS_CID_DISABLED` - The connection ID extension is not included in the client hello, so the DTLS connection ID is not used.
   * :c:macro:`NRF_SO_SEC_DTLS_CID_SUPPORTED` - The connection ID extension with a zero-length CID is included in the client hello, so the modem will accept a DTLS connection ID from the server.
   * :c:macro:`NRF_SO_SEC_DTLS_CID_ENABLED` - The connection ID extension with a valid CID is included in the client hello, so the modem will request DTLS connection ID support.
```

   The default is disabled.

NRF_SO_SEC_DTLS_CID_STATUS
   Get the status of DTLS connection ID.

>    This socket option tells whether the connection ID is used in the current connection.

Reviewer: At the moment, these lines seem to say the same thing. Is it still correct if Line 290 reads "The status tells whether..."?

NRF_SO_SEC_DTLS_CONN_SAVE
   Save DTLS connection.

>    This socket option can be used to pause a session that is not frequently used by the application. Saving the session will free memory in the modem that can be used for other connections.

Reviewer: ```suggestion
   This socket option can be used to pause a session that is not frequently used by the application.
   Saving the session will free memory in the modem, so the memory can be used for other connections.
```
I think it's supposed to say that the memory is what can be used for other connections. If it's the modem that can be used for other connections, apply only the line break.
